pkg/ssh/manager.go

```diff
@@ -217,30 +246,33 @@ func (sm *StreamManager) updateRecordsBinding(ctx context.Context, _ uint64, rec
 	}
 }
 
-func (sm *StreamManager) SetSessionIDForStream(ctx context.Context, streamID uint64, sessionID string, sessionBindingID string) error {
-	sm.mu.Lock()
-	for !sm.initialSyncDone {
-		sm.mu.Unlock()
-		select {
-		case <-sm.waitForInitialSync:
-		case <-time.After(10 * time.Second):
-			return errors.New("timed out waiting for initial sync")
-		case <-ctx.Done():
-			return context.Cause(ctx)
-		}
-		sm.mu.Lock()
+func (sm *StreamManager) OnStreamAuthenticated(ctx context.Context, streamID uint64, req AuthRequest) error {
+	if err := sm.waitForInitialSync(ctx); err != nil {
+		return err
 	}
-	defer sm.mu.Unlock()
-	if sm.sessionStreams[sessionID] == nil {
-		sm.sessionStreams[sessionID] = map[uint64]struct{}{}
+	activeStream := sm.activeStreams[streamID]
+	if activeStream.Session != nil || activeStream.SessionBindingID != nil {
+		return status.Errorf(codes.Internal, "stream %d already has an associated session", streamID)
 	}
```

Comment:
I think this while handling the TODO from EvaluateSSH
> TODO: only do this once, not on re-evaluate

Might cause bugs when there are multiple binding IDs associated to a session.

I think there's a good chance I'm reading the code changes here wrong, but will verify